src/main/java/nl/rijksoverheid/mev/gezagsmodule/domain/gezagvraag/OuderOverledenOfOnbevoegdTotGezag.java

```diff
@@ -28,36 +42,43 @@ protected OuderOverledenOfOnbevoegdTotGezag(final GezagsBepaling gezagsBepaling)
     @Override
     public void perform() {
         Persoonslijst plPersoon = gezagsBepaling.getPlPersoon();
-        answer = V4A_3_JA;
-        
-        // Preconditie: minimaal 1 ouder moet een BSN hebben
-        Ouder1 persoonOuder1 = plPersoon.getOuder1();
-        Ouder2 persoonOuder2 = plPersoon.getOuder2();
-        if ((persoonOuder1 == null || persoonOuder1.getBurgerservicenummer() == null)
-            && (persoonOuder2 == null || persoonOuder2.getBurgerservicenummer() == null)) {
+        var heeftOuder1Burgerservicenummer = plPersoon.getOuder1AsOptional().map(Ouder1::getBurgerservicenummer).isPresent();
+        var heeftOuder2Burgerservicenummer = plPersoon.getOuder2AsOptional().map(Ouder2::getBurgerservicenummer).isPresent();
+        if (!heeftOuder1Burgerservicenummer && !heeftOuder2Burgerservicenummer) {
```

Comment:
deze IF gaat nu fout, de oude code controlleerd of een ouder null is of er een ouder is zonder burgerservice nummer, dus als ouder 1 null / geen burgerservice nummer heeft komt er een exceptie.

De nieuwe code geeft pas een exceptie als zowel ouder 1 als ouder 2 null is of geen burgerservicenummer. dat moet OR zijn

Nitpick : ik vind de leesbaarheid van de oude code een stuk beter.

Reply:
Deze observatie klopt niet.

Uitgangspunt is als volgt: minimaal een ouder moet een burgerservicenummer hebben.

`if (!heeftOuder1Burgerservicenummer && !heeftOuder2Burgerservicenummer) { exceptie }` is als volgt te lezen: als ouder 1 GEEN burgerservicenummer heeft EN ouder 2 OOK GEEN burgerservicenummer heeft, dan exceptie. Dat komt overeen met het uitgangspunt.